src/components/settings/Password.js

```diff
@@ -49,53 +47,63 @@ class Password extends Component<Props, State> {
 
     if (oldPassword && newPassword && newPasswordConfirm) {
       if (newPassword !== newPasswordConfirm) {
-        this.handleMismatchedPassword();
+        this.handleError('mismatchedPassword', true);
+        return false;
+      }
+      if (oldPassword === newPassword) {
+        this.handleError('samePassword', true);
         return false;
       }
       dispatch(changePassword(user.id, oldPassword, newPassword));
     } else {
-      this.handleMissingInput();
+      this.handleError('missingInput', true);
     }
     return true;
   }
 
-  handleMissingInput = () => {
+  handleError = (name, value) => {
     this.setState({
-      missingInput: true,
+      errors: {
+        [name]: value,
+      },
     });
   }
```

Comment:
Please add `;` here and after all arrow functions.
Imagine you would add a normal member to a class like
```
class MyClass {
  member = "Hello!";
}
```
You would add a semicolon there, too. Arrow functions are also just members (because of the `=`).